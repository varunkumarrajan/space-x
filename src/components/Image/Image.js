// React
import React from 'react';
// Prop types
import PropTypes from 'prop-types'; // ES6
// Styled component
import styled from 'styled-components';

// Styled image wrapper
const ImageWrapper = styled.img`
  background-color: #f1f1f1;
  width: 100%;
  max-width: 100%;
  height: auto;
  object-fit: cover;
`;

/**
 * @description Load image
 * @param {object} { content }
 */
const Image = ({ content }) => (
  <ImageWrapper title={content.alt} alt={content.alt} src={content.src} />
);

// Default export
export default Image;

// Default props
Image.defaultProps = {
  content: {
    alt: '',
    src: '',
  }
};

// Prop types
Image.propTypes = {
  content: PropTypes.shape({
    alt: PropTypes.string.isRequired,
    src: PropTypes.string.isRequired,
  }),
};
