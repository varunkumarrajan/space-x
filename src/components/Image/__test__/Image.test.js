// React
import React from 'react';
// Testing library
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';

import Image from './../Image';

// Mock data
const imageMock = {
    alt: 'testing',
    src: 'testing.png'
}

afterEach(cleanup);

describe('Image', () => {
it('Should Render Image With Props', () => {
    const { getByAltText } = render(<Image content={imageMock} />);
    const image = getByAltText('testing');
    expect(image.className).not.toBeFalsy();
    expect(image.title).toContain('testing');
    expect(image.src).toContain('testing.png');
  });

  it('App Snapshot', () => {
    const image = renderer.create(<Image content={imageMock} />).toJSON();
    expect(image).toMatchSnapshot();
  });
});
