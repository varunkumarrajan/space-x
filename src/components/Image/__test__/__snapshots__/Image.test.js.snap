// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image App Snapshot 1`] = `
<img
  alt="testing"
  className="sc-bdfBwQ eVkeNQ"
  src="testing.png"
  title="testing"
/>
`;
